src/sentry/api/endpoints/project_repo_path_parsing.py

```diff
@@ -56,8 +58,9 @@ def org_id(self):
     def validate_source_url(self, source_url: str):
         # first check to see if we are even looking at the same file
         stack_path = self.initial_data["stack_path"]
-        stack_file = stack_path.split("/")[-1]
-        source_file = source_url.split("/")[-1]
+
+        stack_file = PureWindowsPath(stack_path).name
```

Comment:
I don't think this works as expected.
You can probably write unit tests for `PathMappingSerializer` and avoid integration tests for the endpoint. That way you can use parametrization.

```python
from pathlib import PureWindowsPath
stack_path = "../foo.py"
print(PureWindowsPath(stack_path).name)
stack_path = "/c/d/foo.py"
print(PureWindowsPath(stack_path).name)
stack_path = "c:\Windows\foo.py"
print(PureWindowsPath(stack_path).name)
stack_path = "c:\\Windows\\foo.py"
print(PureWindowsPath(stack_path).name)
```

```bash
$> python paths.py
foo.py
foo.py
Windows
       oo.py
foo.py
```

Reply:
@armenzg `"c:\Windows\foo.py"` is not valid, a single `\` is an escape character, not `\` character. That's why it has to be escaped as `"\\"`

```
>>> stack_path = "c:\\Windows\\foo.py"
>>> print(stack_path)
c:\Windows\foo.py
```

Reply:
@armenzg `"\\"` is a single backslash
`"\"` is an `SyntaxError`